tutorials/text_processing/WFST_Tutorial.ipynb

```diff
@@ -0,0 +1,8368 @@
+      ]
+     },
+     "execution_count": 412,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "\"\"\"\n",
+    "# ERROR IF RUN\n",
+    "digits = pynini.string_file(\"data/digits.tsv\")\n",
+    "\n",
+    "\n",
+    "# If utils.py is in working directory you may also use `get_abs_path`, which will\n",
+    "# always call paths relative to your {LANGUAGE} directory.\n",
+    "from nemo_text_processing.inverse_normalization.{LANGUAGE}.utils import get_abs_path\n",
+    "digits = pynini.string_file(get_abs_path(\"data/digits.tsv\"))\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "yPccmicQkYAB"
+   },
+   "source": [
+    "While we will use `string_map` throughout this tutorial, please note that NeMo employs the later option for maintainability and recommends its use instead."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "FQJiJcVMrNmC"
+   },
+   "source": [
+    "### Teens\n",
+    "\n",
+    "Let us consider our next set of numbers:\n",
+    "- 10 - dix\n",
+    "- 11 - onze\n",
+    "- 12 - douze\n",
+    "- 13 - treize\n",
+    "- 14 - quatorze\n",
+    "- 15 - quinze\n",
+    "- 16 - seize\n",
+    "- 17 - dix-sept\n",
+    "- 18 - dix-huit\n",
+    "- 19 - dix-neuf\n",
+    "\n",
+    "Lke before, we can simply use `string_map` to compose a WFST for them. But note how there is some redundancy in the number set: `17`, `18`, and `19` are all of the form `dix + digit`. It would be more efficient simply to reuse our prior WFST in these cases than simply creating new arcs, states, and weights. \n",
+    "\n",
+    "We can achieve this using pynini's string concatenation function to extend the accepted input strings. First we will create an WFST for `11-16`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 413,
+   "metadata": {
+    "id": "orSgBwyXsfY5"
+   },
+   "outputs": [],
+   "source": [
+    "teens = pynini.string_map([\n",
+    "\t\t\t\t(\"onze\",\"11\"),\n",
+    "\t\t\t\t(\"douze\",\"12\"),\n",
+    "\t\t\t\t(\"treize\",\"13\"),\n",
+    "\t\t\t\t(\"quatorze\",\"14\"),\n",
+    "\t\t\t\t(\"quinze\",\"15\"),\n",
+    "\t\t\t\t(\"seize\",\"16\"),\n",
+    "])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "s1yIgigdtriQ"
+   },
+   "source": [
+    "Now, we will create a `tens` WFST that is responsible for mapping all instances of \"dix\" and concatenate it with the prior `digits` WFST. (Deleting any possible hyphens in-between.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 414,
+   "metadata": {
+    "id": "CzwZrFCkt87W"
+   },
+   "outputs": [],
+   "source": [
+    "tens = pynini.string_map([(\"dix\", \"1\")])\n",
+    "delete_hyphen = pynutil.delete(\"-\").ques # equivalent to deleting /-?/\n",
```

Comment:
this will also allow invalid numbers like `dix-trois` instead of treize? if so could you mention this?

Reply:
I bring it up later in the section, would it be better earlier?